src/Elasticsearch.Net/Transport/Pipeline/ResponseBuilder.cs

```diff
@@ -76,29 +76,25 @@ private void SetBody(ElasticsearchResponse<TReturn> response, Stream stream)
 			var needsDispose = typeof(TReturn) != typeof(Stream);
 			using (needsDispose ? stream : EmptyDisposable)
 			{
-				if (response.Success)
+				if (response.Success || response.AllowAllStatusCodes)
 				{
 					if (!SetSpecialTypes(stream, response, bytes))
 					{
 						if (this._requestData.CustomConverter != null) response.Body = this._requestData.CustomConverter(response, stream) as TReturn;
 						else response.Body = this._requestData.ConnectionSettings.Serializer.Deserialize<TReturn>(stream);
 					}
+					if (response.AllowAllStatusCodes) 
+						ReadServerError(response, new MemoryStream(bytes), bytes);
```

Comment:
`MemoryStream` does not hold managed resources and will get gc'd just like the `byte[]` it wraps.

That said it will be disposed down the line like network streams would be too when they are deserialized (see ServerError.Create(Async).